app/models/post.rb

```diff
@@ -0,0 +1,9 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id', counter_cache: true
+  has_many :likes
+  has_many :comments
+
```

Comment:
- Based on the requirement, you should have a method that updates the posts counter for a user, please check the suggestion below and update your model 👍🏿 


```suggestion
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

-  [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save can be used to call the #update_post_counter method every time a new post has been saved to database, you need the  update_post_counter method private